autogpts/autogpt/autogpt/core/resource/model_providers/openai.py

```diff
@@ -396,6 +396,64 @@ def count_message_tokens(
         num_tokens += 3  # every reply is primed with <|start|>assistant<|message|>
         return num_tokens
 
+    async def create_chat_completion_raw(
+        self,
+        model_prompt: list[ChatMessage],
+        model_name: OpenAIModelName,
+        functions: Optional[list[CompletionModelFunction]] = None,
+        **kwargs,
+    ) -> AssistantChatMessage:
+        """Create a completion using the OpenAI API."""
```

Comment:
`create_chat_completion_raw` is similar to `create_chat_completion` but doesn't parse the response. I kept `create_chat_completion` for backwards compatibility a it's used across the project.

Reply:
So using `create_chat_completion` without specifying `completion_parser` would be equivalent to using `create_chat_completion_raw`?

I don't see a reason to create a separate method with less functionality than the existing one. It's quite handy to have a shortcut to parsing and retrying generation if parsing fails, not just for the agent. Why do you think this functionality shouldn't be in the LLM provider?

Reply:
How about we pass schema object instead of entire parsing function (now it's only `JSONSchema` could be others), this way we retain retry and parsing but it's easier.

Reply:
That would both limit what can be implemented using the parse-fix mechanism and increase the complexity of the `create_chat_completion` implementation beyond its scope.

I'm not sure what your issue was, but maybe we can resolve it by making it easier to use a `PromptStrategy` combined with a `ChatModelProvider`. E.g.:
```py
provider = OpenAIProvider()
ai_block = SomePromptStrategy(provider)
result = ai_block.invoke(3, input_arg_2="Hello, World!")
```
In that case we could move the parsing and retrying logic to the `PromptStrategy.invoke` method.

Reply:
I'll just use `create_chat_completion` in this PR, I noted that and might try this in a separate PR.